x-pack/plugins/cases/public/components/files/file_download_button_icon.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiButtonIcon } from '@elastic/eui';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+
+import type { Owner } from '../../../common/constants/types';
+
+import { constructFileKindIdByOwner } from '../../../common/constants';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import * as i18n from './translations';
+
+interface FileDownloadButtonIconProps {
+  fileId: string;
+}
+
+const FileDownloadButtonIconComponent: React.FC<FileDownloadButtonIconProps> = ({ fileId }) => {
+  const { owner } = useCasesContext();
+  const { client: filesClient } = useFilesContext();
+
+  return (
+    <EuiButtonIcon
+      iconType={'download'}
+      aria-label={i18n.DOWNLOAD}
+      href={filesClient.getDownloadHref({
```

Comment:
I noticed through the reviews of the files PRs that we use this pattern a lot. Should we extract the logic of the construction of the `href` to another function? For example:

```
const getFileDownloadHref = ({ filesClient, fileId, owner }) => filesClient.getDownloadHref({
        fileKind: constructFileKindIdByOwner(owner[0] as Owner),
        id: fileId,
      })
```

Reply:
I can but I don't see much difference between

`filesClient.getDownloadHref(A, B)`

and

`newFunction(filesClient.getDownloadHref, A, B)` 😄

Reply:
The only difference is this Cases logic: `fileKind: constructFileKindIdByOwner(owner[0] as Owner)` is replicated over and over again. If we need to change it in the future we have to go to all places to do it. I am fine with leaving it as it is. Not a big deal. I get your point.